google-pubsub.yml

```diff
@@ -34,6 +34,64 @@ provision:
     constraints:
       maximum: 600
       minimum: 10
+  - field_name: topic_message_retention_duration
+    type: string
+    details: |
+      Duration to retain messages in the topic if no subscription is attached. (e.g., "604800s" for 7 days)
+    default: ""
+  - field_name: topic_kms_key_name
+    type: string
+    details: |
+      KMS key name for encryption.
+      The expected format is `projects/*/locations/*/keyRings/*/cryptoKeys/*`
+    default: ""
+  - field_name: subscription_message_retention_duration
+    type: string
+    details: |
+      Duration to retain unacknowledged messages in the subscription.
+      If `subscription_retain_acked_messages` is true, then this also configures the retention
+      of acknowledged messages, and thus configures how far back in time a subscriptions.seek can be done.
+    default: ""
+  - field_name: subscription_retain_acked_messages
+    type: boolean
+    details: |
+      Retain acknowledged messages.
+      If true, then messages are not expunged from the subscription's backlog, even if they are acknowledged,
+      until they fall out of the messageRetentionDuration window.
+    default: false
+  - field_name: subscription_expiration_policy
+    type: string
+    details: |
+      Duration after which the subscription expires if no messages are received (e.g., "604800s" for 7 days)
+      If expirationPolicy is not set, a default policy with ttl of 31 days will be used.
+      If it is set with "", the resource never expires. The minimum allowed value is 1 day
```

Comment:
These two sentences seem to be contradicting. 

> If expirationPolicy is not set, a default policy with ttl of 31 days will be used. 
> If it is set with "", the resource never expires.

So when the customer does not explicitly set the `subscription_expiration_policy` we will default to `""` which means we don't pass anything to tofu so it defaults to 31 days.

If the customer passes explicitly  `""` we don't pass anything to tofu so it defaults to 31 days and not to never expire.

Reply:
Great catch!! Thank you very much @FelisiaM . I am fixing it now

Reply:
Fixed! Notice that once the value is set, we can no longer unset it!